### DISCUSSION TITLE
なんかリアクションが別のノートに吸い取られている（RBTが原因化か？）

### DISCUSSION THREAD

**Discussion by mendakon** [misskey-dev/misskey#14770] [General]
https://sushi.ski にて
RBTをONにしたところ、

![image](https://github.com/user-attachments/assets/86d293c0-9983-49e9-8153-a7c92498ef77)
本来何個か絵文字がついていたはずのノート

![image](https://github.com/user-attachments/assets/def6e26a-48ae-42da-90d1-e4b1b66fd580)
20個もついているはずのないノート

ある時から、前日にかけてのノートのリアクション数が最近のノートのリアクション数に吸い取られているような挙動をする
一定範囲のノートのリアクションは消え、その範囲の終端あたりのノートに存在しない量のリアクションがついているように見える

なお、この事象をかくにんしてから**RBTをOFFにしてredisのキャッシュを全パージしたが事象は改善されなかった**

データベースを見たところリアクションは正常にデータベースに反映されているようなので取得部分の問題？

misskeyの構造によく詳しくないので調査をお手伝いしていただきたい

**Comment by mendakon** [comment]
![image](https://github.com/user-attachments/assets/449b8b0e-b878-4c8e-be59-9e9295c1799e)

下２つのリアクションが上２つのノートに吸い取られている？様子。明らかに上の方の絵文字数が異常
meow_nodは最近ついたリアクションなので反映されているがもっとリアクションがあったはず

**Comment by mendakon** [comment]
[@kuchibashi@sushi.ski](https://sushi.ski/@kuchibashi)

と思ったけど2日前の投稿から全部リアクション消えてるな
吸い取られているどころではない

ほとんどのユーザーで同じ事象が起きてるっぽい

**Comment by tkmrgit** [comment]
Reaction Boost TechnologyじゃなくてReaction Absorb Technologyだった...?

**Comment by mendakon** [comment]
参考
https://sushi.ski/notes/9zfeb2uv6j
![image](https://github.com/user-attachments/assets/51ee0d4c-9403-4e78-a412-b31f37ea7274)

なのにDB上にはある。
```
mk1=# SELECT "noteId", reaction FROM note_reaction WHERE "noteId" = '9zf9vm8qns';
   noteId   |         reaction
------------+--------------------------
 9zf9vm8qns | :ore_:
 9zf9vm8qns | :ore_:
 9zf9vm8qns | :nani_ga_hajimarundesu:
 9zf9vm8qns | :ore_:
 9zf9vm8qns | :ore_:
 9zf9vm8qns | :nani_ga_hajimarundesu:
 9zf9vm8qns | :ore:
 9zf9vm8qns | :ore_:
 9zf9vm8qns | :ore_:
 9zf9vm8qns | :ore_:
 9zf9vm8qns | :ore_:
 9zf9vm8qns | :ore_:
 9zf9vm8qns | :nani_ga_hajimarundesu:
 9zf9vm8qns | :ore:
 9zf9vm8qns | :ore_:
 9zf9vm8qns | :hello_catsushimgr:
 9zf9vm8qns | :ore:
 9zf9vm8qns | :seikin_furuchara_dance:
 9zf9vm8qns | :helve_hi@tanoshii.site:
 9zf9vm8qns | :ore_:
 9zf9vm8qns | :ian:
 9zf9vm8qns | :ore_:
 9zf9vm8qns | :nani_ga_hajimarundesu:
(23 rows)
```

**Reply by anatawa12** [reply]
表示用のリアクションはnoteテーブルのjsonb型のreactionsが使用されているのでこっちのデータが壊れていると考えられます

**Reply by mendakon** [reply]
ほんまや！ありがとうございます

```
mk1=# SELECT * FROM note WHERE "id" = '9zfeb2uv6j';
-[ RECORD 1 ]------------+-------------------
id                       | 9zfeb2uv6j
replyId                  |
renoteId                 |
text                     | コンビニ行こうかな
name                     |
cw                       |
userId                   | 8hn59vf3vh
localOnly                | f
renoteCount              | 0
repliesCount             | 0
reactions                | {}
visibility               | public
uri                      |
fileIds                  | {}
attachedFileTypes        | {}
visibleUserIds           | {}
mentions                 | {}
mentionedRemoteUsers     | []
emojis                   | {}
tags                     | {}
hasPoll                  | f
userHost                 |
replyUserId              |
replyUserHost            |
renoteUserId             |
renoteUserHost           |
url                      |
channelId                |
threadId                 |
reactionAcceptance       |
clippedCount             | 0
reactionAndUserPairCache | {}
```

**Comment by anatawa12** [comment]
これ放置しちゃってるけどなにかしらバグってるのは確定しそうなのでissueにしたい

けど再現性が低いらしい？ https://misskey.niri.la/notes/a0j1xg1hlt

**Reply by fruitriin** [reply]
これって再現性あるんです？(うちの鯖では起きてないような気がして

**Comment by anatawa12** [comment]
RBT有効にしてるにりみすでも発生してるのでなんかしらあるかもしれないです

にりみすでこのノート以外に過去に数件見た記憶があります

<img width="231" height="236" alt="image" src="https://github.com/user-attachments/assets/645c010b-15c1-4aed-ae9b-c0e0c97fe6a7" />

https://misskey.niri.la/notes/akyn5m2x61
